src/components/pages/UserPage.js

```diff
@@ -39,10 +39,7 @@ export const UserPage = ({ match }) => {
       }),
     );
 
-    // clone previous userRepos and append new key value pair
-    const newUserRepos = Object.assign({}, userRepos);
-    Object.assign(newUserRepos, { [user]: repoData });
-    setUserRepos(newUserRepos);
+    setUserRepos({ ...userRepos, [user]: repoData });
```

Comment:
if I remember correctly, there's a possible bug that this slows down due to the destructuring nature. Can you try looking up 20 usernames or so?

if the performance doesn't seem to drop i this change should be fine